File: Regresion lineal simple/simple_linear_regression.py
# -*- coding: utf-8 -*-
"""
Created on Sat Mar 20 19:44:52 2021

@author: axvargas
"""
#lybraries import
import numpy as np
import pandas as pd 

#data import
dataset = pd.read_csv('Salary_Data.csv')

X = dataset.iloc[:,:-1].values
y = dataset.iloc[:,1].values


#divide dataset in training and testing
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=1/3, random_state=0)

#scaling variables
## EN EL CASO DE LA REG LINEAL SIMPLE NO ES NECESARIO

#create linear regression model
#The library does the dirty work
from sklearn.linear_model import LinearRegression
regression = LinearRegression()
regression.fit(X_train,y_train)

#Predict test set
y_pred = regression.predict(X_test)

#visualize the training data
import matplotlib.pyplot as plt

plt.scatter(X_train,y_train, c="red")
plt.plot(X_train, regression.predict(X_train), c="blue")
plt.title("Sueldo vs Años de Experiencia(Conjunto de Entrenamiento)")
plt.xlabel("Años de Experiencia")
plt.ylabel("Sueldo($)")
plt.show()

#visualize the testing data
plt.scatter(X_test,y_test, c="red")
plt.plot(X_train, regression.predict(X_train), c="blue") #Se puede dejar el de entrenamiento ya que es la misma recta
plt.title("Sueldo vs Años de Experiencia(Conjunto de Test)")
plt.xlabel("Años de Experiencia")
plt.ylabel("Sueldo($)")
plt.show()

File: Regresion lineal simple/simple_linear_regression.R
#Regresion lineal simple

#import dataset
dataset = read.csv("Salary_Data.csv")


#divide dataset into training and testing
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split==TRUE)
testing_set = subset(dataset, split==FALSE)

#Ajustar el modelo de regresion lineal simple con el conjunto de entrenamiento
regressor=lm(formula = Salary ~ YearsExperience,
            data = training_set)
summary(regressor)

#Predecir resultados con el conjunto de test
y_pred = predict(regressor, newdata = testing_set) # las columnas deben llamarse igual que en training set

#Visualizar los datos del conjunto de entrenamiento
library(ggplot2)
ggplot() +
  geom_point(aes(x=training_set$YearsExperience, y=training_set$Salary),
             colour="red")+
  geom_line(aes(x=training_set$YearsExperience,
                y=predict(regressor, newdata = training_set)),
            colour="blue")+
  ggtitle("Sueldo vs Años de Experiencia (Conjunto de Entrenamiento)")+
  xlab("Años de Experiencia")+
  ylab("Sueldo ($)")

#Visualizar los datos del conjunto de testeo
ggplot() +
  geom_point(aes(x=testing_set$YearsExperience, y=testing_set$Salary),
             colour="red")+
  geom_line(aes(x=training_set$YearsExperience,
                y=predict(regressor, newdata = training_set)), #Se puede dejar los datos de entrenamiento ya que es la misma recta
            colour="blue")+
  ggtitle("Sueldo vs Años de Experiencia (Conjunto de Testeo)")+
  xlab("Años de Experiencia")+
  ylab("Sueldo ($)")
